lib/queue.js

```diff
@@ -46,255 +41,242 @@ FetchQueue.prototype.add = function(protocol, domain, port, path, depth, callbac
+        return callback(new Error("Invalid statistic"));
     }
 
-    self.forEach(function(item) {
-        if (item.fetched && item.stateData[statisticName] !== null && item.stateData[statisticName] > maxStatisticValue) {
-            maxStatisticValue = item.stateData[statisticName];
+    queue.forEach(function(item) {
+        if (item.fetched && item.stateData[statisticName] > maximum) {
+            maximum = item.stateData[statisticName];
         }
     });
 
-    callback(null, maxStatisticValue);
-    return maxStatisticValue;
+    callback(null, maximum);
 };
 
 // Gets the minimum total request time, request latency, or download time
 FetchQueue.prototype.min = function(statisticName, callback) {
-    callback = callback && callback instanceof Function ? callback : function() {};
-    var minimum,
-        minStatisticValue = Infinity,
-        self = this;
-
-    if (allowedStatistics.join().indexOf(statisticName) === -1) {
-        // Not a recognised statistic!
-        return callback(new Error("Invalid statistic."));
+    callback = callback instanceof Function ? callback : noop;
+
+    var minimum = Infinity,
+        queue = this;
+
+    if (allowedStatistics.indexOf(statisticName) === -1) {
+        return callback(new Error("Invalid statistic"));
     }
 
-    self.forEach(function(item) {
-        if (item.fetched && item.stateData[statisticName] !== null && item.stateData[statisticName] < minStatisticValue) {
-            minStatisticValue = item.stateData[statisticName];
+    queue.forEach(function(item) {
+        if (item.fetched && item.stateData[statisticName] < minimum) {
+            minimum = item.stateData[statisticName];
         }
     });
 
-    minimum = minStatisticValue === Infinity ? 0 : minStatisticValue;
-    callback(null, minimum);
-    return minimum;
+    callback(null, minimum === Infinity ? 0 : minimum);
 };
 
 // Gets the minimum total request time, request latency, or download time
 FetchQueue.prototype.avg = function(statisticName, callback) {
-    callback = callback && callback instanceof Function ? callback : function() {};
-    var average,
-        NumberSum = 0,
-        NumberCount = 0,
-        self = this;
-
-    if (allowedStatistics.join().indexOf(statisticName) === -1) {
-        // Not a recognised statistic!
-        return callback(new Error("Invalid statistic."));
+    callback = callback instanceof Function ? callback : noop;
+
+    var sum = 0,
+        count = 0,
+        queue = this;
+
+    if (allowedStatistics.indexOf(statisticName) === -1) {
+        return callback(new Error("Invalid statistic"));
     }
 
-    self.forEach(function(item) {
-        if (item.fetched && item.stateData[statisticName] !== null && !isNaN(item.stateData[statisticName])) {
-            NumberSum += item.stateData[statisticName];
-            NumberCount++;
+    queue.forEach(function(item) {
+        if (item.fetched && Number.isFinite(item.stateData[statisticName])) {
```

Comment:
I was unsure of node 0.10 support at first as well, thought I'd rely on the CI to see how it did, and it passed. That's what the module requires as the lowest supported version in the package.json as well, so I figure it's no problem if we make use of it. Whatcha think?